File: careerpilot-backend/src/main/java/com/careerpilot/service/CodingService.java
package com.careerpilot.service;

import com.careerpilot.dto.CodingQuestionDto;
import com.careerpilot.dto.CodingSubmissionDto;
import com.careerpilot.dto.SubmitCodeRequest;
import com.careerpilot.entity.Difficulty;

import java.util.List;

public interface CodingService {

    List<CodingQuestionDto> getQuestions(Difficulty difficulty, String search);

    CodingQuestionDto getQuestionById(Long id);

    CodingSubmissionDto submitCode(SubmitCodeRequest request, Long userId);

    List<CodingSubmissionDto> getMySubmissions(Long userId);

    List<CodingSubmissionDto> getSubmissionsByQuestion(Long questionId, Long userId);
}

File: careerpilot-frontend/src/services/codingService.js
import axiosInstance from './axiosInstance';

const codingService = {
  getQuestions: (difficulty, search) => {
    const params = {};
    if (difficulty) params.difficulty = difficulty;
    if (search)     params.search     = search;
    return axiosInstance.get('/api/coding/questions', { params });
  },

  getQuestionById: (id) => axiosInstance.get(`/api/coding/questions/${id}`),

  submit: (payload) => axiosInstance.post('/api/coding/submit', payload),

  getMySubmissions: () => axiosInstance.get('/api/coding/submissions'),

  getSubmissionsByQuestion: (questionId) =>
    axiosInstance.get(`/api/coding/questions/${questionId}/submissions`),
};

export default codingService;
